sycl/include/sycl/detail/generic_type_traits.hpp

```diff
@@ -719,17 +721,14 @@ struct RelConverter<T,
   }
 };
 
-template <typename T>
-struct RelConverter<T,
-                    typename std::enable_if_t<!TryToGetElementType<T>::value>> {
-  using R = internal_rel_ret_t<T>;
+template <> struct RelConverter<bool, void> {
 #ifdef __SYCL_DEVICE_ONLY__
-  using value_t = bool;
+  using R = bool;
 #else
-  using value_t = R;
+  using R = bool;
```

Comment:
Seems like me might as well just remove the branching here. Maybe even remove `R` and just inline it below. Do we use it outside the struct?